models/user.js

```diff
@@ -23,7 +23,12 @@ const MODEL = {
         .string()
         .max(128)
         .description('The SCM to which the user belongs')
-        .example('github:github.com')
+        .example('github:github.com'),
+
+    settings: Joi
+        .object()
+        .description('User settings')
+        .example({ display: { jobNameLength: 20 } })
```

Comment:
I was thinking for user preference schema like: 

```
settings: {
   pipelines: [{
	 id: 1, // or pipelinId, either way
   	 jobNameLength: 20
   }, {
	 id: 2,
   	 jobNameLength: 30
   }]
   x: 'y', // this can be used later as global settings, rather than pipelines specifics
}
```

Reply:
I guess it depends if you think pipeline settings should be personalized per user or per pipeline
like repo settings in Github are tied to the repo, not the user